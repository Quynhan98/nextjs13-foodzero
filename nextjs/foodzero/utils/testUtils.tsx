import { ReactNode } from 'react'
import { render } from '@testing-library/react'
import { ChakraProvider } from '@chakra-ui/react'

// Themes
import { customTheme } from '@themes/index'

// Contexts
import { AuthProvider } from '@contexts/AuthProvider'
import { BookingProvider } from '@contexts/BookingProvider'

export const customRender = (component: ReactNode) => {
  return render(
    <ChakraProvider theme={customTheme}>
      <AuthProvider>
        <BookingProvider>{component}</BookingProvider>
      </AuthProvider>
    </ChakraProvider>,
  )
}

export * from '@testing-library/react'
export * from '@testing-library/jest-dom'
export { customRender as render }
