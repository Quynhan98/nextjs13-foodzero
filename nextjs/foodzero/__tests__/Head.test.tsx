// Utils
import { render } from '@utils/testUtils'

// App
import Head from 'app/head'

describe('Head render', () => {
  it('Should Head match Snapshot', () => {
    const { container } = render(<Head />)

    expect(container).toMatchSnapshot()
  })
})
