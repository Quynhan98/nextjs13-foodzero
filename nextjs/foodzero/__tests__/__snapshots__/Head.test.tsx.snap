// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Head render Should Head match Snapshot 1`] = `
<div>
  <title>
    Foodzero
  </title>
  <meta
    content="width=device-width, initial-scale=1.0"
    name="viewport"
  />
  <meta
    content="Foodzero"
    name="title"
  />
  <meta
    content="This is a food restaurant"
    name="description"
  />
  <meta
    content="website"
    property="og:type"
  />
  <meta
    content="Foodzero"
    property="og:title"
  />
  <meta
    content="This is a food restaurant"
    property="og:description"
  />
  <meta
    property="og:url"
  />
  <meta
    content="summary"
    property="twitter:card"
  />
  <meta
    content="Foodzero"
    property="twitter:title"
  />
  <meta
    content="This is a food restaurant"
    property="twitter:description"
  />
  <link
    href="/images/logo.webp"
    rel="icon"
  />
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
